### ISSUE TITLE
Stabilize CRW Site

### ISSUE THREAD

**Issue by turnrye** [memhamwan/proposals#1]
# Background

MemHamWAN has one point-of-presence in the Memphis medical district named "LEB", a point-of-presence in east Memphis named "HIL", and a point-of-presence in Olive Branch named "CRW". Today there are interconnections between LEB<->HIL and LEB<->CRW. The OBARC is wishing to use local network connectivity at CRW to host repeater infrastructure. The OBARC is also wishing to get network connectivity at a new location in Lewisburg, which we'll refer to as "LWA". MemHamWAN has no lease today at LWA. The lease at CRW dictates the configuration of the antennas, and a substantial change to the mast or antennas would require amending the lease.

# Problems

There are three problems today related to MemHamWAN's expansion into DeSoto county, and with local amateurs there wishing to bring repeater infrastructure online soon, these issues become more urgent to address. While each of these can be addressed individually, in totality they represent the actions needed to bring reliable network connectivity to both the CRW and LWA sites for OBARC to be able to bring repeaters online.

## Connectivity at CRW

The point-to-point from CRW to LEB that CRW depends on is unstable. Multiple attempts at aiming both ends of the link have taken place and have only yielded marginal improvements. At this point, the link is unable to sustain a reliable connection. It needs to be improved. At this point, the CRW site is a terminal node in the network. Despite a few surveys in the past, we've not found much opportunity to serve clients off of it directly. The site has multiple IP cameras and an Ethernet port that has been offered to OBARC to run networked repeaters off of, plus a server that has never been used substantially. The CRW site has no sector antennas -- only an omni point-to-multi-point for subscribers.

## Connecting LWA to HamWAN

LWA is a site that does not have a clear opportunity at this point for HamWAN aside from linking repeater infrastructure in partnership with OBARC. There may be further expansion opportunities to the south in the future, but until there are clearer criteria for these, a solution cannot be designed. LWA was originally (September 2022) being considered for network expansion, but HamWAN has failed to secure a lease or plan an install. The previously proposed HamWAN solution has proven to be overly complex for what the organization can support currently. The solution is also misaligned with what the community needs -- the value to OBARC is in getting network connectivity at the repeater site, not deploying a cell site.

## Power at CRW

Currently at CRW, there is an abandoned PDU that will not auto-start. There is also a Tripp Lite UPS that seems to not be functioning properly – it is not auto-starting cold or after charging from dead. Between both of these devices, there have been extended periods of time where the equipment is simply not powered on. Due to the connectivity problems, these have gone unobserved.

Also, the site suffers power disruptions more frequently than our other sites, and these outages appear to be longer in nature than others, often leading to exceeding the capacity of our UPS. This leads to more frequent charge-from-dead situations for the UPS and cold starts. This process seems to be less reliable compared to normal UPS operations with brief outages.

# Solution

## Connectivity at CRW

A simple (i.e. not requiring a long chain of site visits) fix to this problem would be to re-aim the dish at CRW to point to our HIL site in East Memphis. The distance is shorter and the sector antenna and narrower 10 MHz channel width would certainly resolve any aiming issues. This link would only support about 15 MBit of bandwidth, but this would support our use cases as long as we maintain only 2 IP cameras at the site. Stations linking off of CRW's omni wouldn't have near as much bandwidth, but if we found that this is a problem, we could in the future resolve this by installing a point-to-point dish at the HIL side of the network.

## Connectivity at LWA

Deploying LWA as a client node under OBARC's ownership would simplify things greatly, and so that approach is recommended. At this point, configuring LWA as a client node in partnership with OBARC makes the most sense. It should connect directly to HIL and could be brought online by that group independently of the efforts to improve CRW. 

### Donations to OBARC

These items would make for a client node that can be preconfigured by @turnrye and then donated to OBARC. It would be setup as a managed client node so that HamWAN can help provide remote administration of the router and OBARC would also be able to manage the device directly. This would allow OBARC to deploy the hardware under their own lease and liability insurance independent of HamWAN's actions at CRW. All of these devices are already held in inventory and without an assigned use -- it would not require any new purchases for either organization.

- Poynting K-GRID-003-06 30 dBi grid dish
- Metal 5SHPn with hoseclamp style mount
- CAT5e shielded cabling, preterminated at 50m length
- Ethernet lightning arrester and grounding for installing at the shack entrypoint
- Passive gigabit power injector and 24VDC power supply

OBARC would need to provide a switch at their shack if they intend to connect multiple devices to the network.

This is illustrated below, also noting a client node to be installed at the Lewisburg water tower in partnership with OBARC.

![Untitled Diagram drawio(1)](https://user-images.githubusercontent.com/701035/219960450-94b52aa0-adeb-4120-ac4c-1f8ebc3cfe60.svg)

Below is a map showing both the existing and the new proposed RF connections. Lines are labelled with their length in KM. 

![export](https://user-images.githubusercontent.com/701035/219961271-ac8bb709-2351-4bc9-990d-e9320c2685b6.png)

Below is a diagram showing the profile between LWA and HIL (the right hand side is HIL, and there the building is 100m tall; the left hand side is LWA, and there the structure is estimated to be between 30m and 40m tall, depending on the mounting location).
![profile-lwa-to-hil](https://user-images.githubusercontent.com/701035/219972753-7ce40e5e-14bc-470d-b81e-5f2eab23003d.png)

Below is a diagram showing the RF link potential between LWA and HIL.
<img width="965" alt="Screen Shot 2023-02-19 at 1 50 15 PM" src="https://user-images.githubusercontent.com/701035/219972941-4eae9d78-6a57-41e0-b3df-b350c39ee153.png">

Below is a diagram showing the profile between HIL and CRW (the right hand side is CRW). It has previously been observed that HIL is clearly distinguishable from the top of the water tower by the eye.

![profile-hil-to-crw](https://user-images.githubusercontent.com/701035/219973138-76acc7c2-abd0-40cd-8644-3cf3a9cdc233.png)

Link path analysis was not done between CRW and HIL as the link is considered assured given the closeness of the link.

## Power at CRW

For the CRW power situation, replace this with a SmartUPS that has a network management card. This would provide more reliable backup power and remote management capabilities in line with most of our existing setups.

For the loads, there are two categories at this site: critical and non-critical. Critical loads are for the core network services: routing, switching, linking to other sites, and point-to-multi-point. The non-critical loads include the server, IP cameras, and administrative WiFi interface. Luckily all of the critical loads can be powered from the cabinet using a single 24V passive POE connection. As a result, the solution to provide redundancy to the critical loads can simply be a 24VDC passive redundant setup. The RB3011 installed at the site as two power options: POE in on ether1 and a dedicated barrel connector on the back. The router will automatically use whichever port has higher voltage and seamlessly fail over in case on of the two goes away.

To provide redundancy to the power in case of future UPS issues, power the UPS both via the jack in the back _and_ the ether 1 POE in. This allows the device to negotiate either of the power supplies. Port 10 on the router is POE out, which would have this redundancy built in -- use this jack for connecting and powering the PowerBox on the top that is running the critical loads (radios, not cameras). Use existing power supplies out of inventory for this portion of the work and just ensure that one side is plugged into the UPS and one side is bypassing the UPS.

## New purchases (un-budgeted)

This change would require the following expenditures to retrofit the existing point-to-point radio into a point-to-multi-point client radio at CRW. No new equipment would be needed at other HamWAN sites.

- [ ] [RB912UAG-5HPnD-OUT](https://www.streakwave.com/mikrotik-rb912uag-5hpnd-out-5ghz-basebox5-802-11bgn-outdoor) - $73.42 + S&H - specifically NOT the US version, needs to be INTL version - this is the replacement radio for the PtP link because the existing one won't do 10 MHz channel widths or ham bands
- [ ] [RF Elements StationBox ALU IP55](https://www.ispsupplies.com/RF-Elements-SBX-ALU-6N1E) - $41 + S&H - this is a replacement case for the one that comes with the basebox above to provide shielding from local interference.

Total expense estimated at (73.42+41)*1.4=$160.19

## New purchases - previously budgeted

We have in inventory an APC SMT2200RM2U Smart-UPS that is missing the following components to be ready for deployment:

- [x] [APC 911-0028E battery tray and RBC43 batteries](https://www.ebay.com/itm/285073899545?epid=1140513948&hash=item425fba8019:g:-VMAAOSwL21jkSia&amdata=enc%3AAQAHAAAAoGIPVMdSD58h9k60xvjl93u5HdN6f%2F%2FjN2gGcBRvL93cOHfGmGS0FfeixInIAa3FHdKmTAf7hP5TLntbixMUQQlYKB99PrisuSvfSSh8rvrUu0jW4%2B1E8YJP%2BAkJh12Yz%2FAcn7cyJzKhrm0AFXS9CrchywCPMEvYEYnqcvh8JM1Oo%2F65DHBPT3xsRxqJwUYMx26IWtfb42pR2Ocfn%2BE4o4I%3D%7Ctkp%3ABk9SR6TM7q_NYQ) @ $149.95
- [x] [rail kit](https://www.ebay.com/itm/374383507741?epid=1200499285&hash=item572afef91d:g:y0wAAOSwWa9jiS9q&amdata=enc%3AAQAHAAAAoJ%2BRU4MVYSLuFE6rGdiMZuq7SflPZ5eWlmKypAhkoFakmASlJh0UnvJDj5dbfiquf72KwgwVUZ5pN0oq6POSMO9JaEJjsoSwWGr2Oa5QfIcvRt0h8DxD%2Bi7L18YRyA0O4HPXjL1LA2IznyRqi9A8y71ohvioiW2TWK0Y86gaYyZHVLZDiNhcTs%2B1ozZFm7e61uZ960o6dRUR8SnBjAMVdz8%3D%7Ctkp%3ABk9SR_Cmh7DNYQ) @ $39.99
- [x] [AP9640 network management card](https://www.ebay.com/itm/314286108182?epid=22036437138&hash=item492ce96e16:g:NgAAAOSwZVBjo714&amdata=enc%3AAQAHAAAA0OKKtnCNWjFPZnR9mcsnbhuds%2B%2Buo99CzpklHzJUy2Tmc51UJFjR1f16csABQ2%2BEFCzO6XVp7E1TfIrrXAmgMZeDcGeb7vIs%2FMwKz18UTeKq0shEM8Are313nCM%2Fj6t2G2LU%2FeQNwI23cDjLevMCxRGuCOynbqbn8xv19lQo775Oz5rOe3sEOUKcjeC3d2SPivmLiswOuAW8ohvntXYMArZFJkbSdqqDlkN0jNpICReZ2ugYM1DrUnP%2Fx%2BQ3F0VFsimn%2BVQe56sgyXww4kCv728%3D%7Ctkp%3ABFBM-OPbr81h) @ $173
- [x] [Rack ears](https://www.ebay.com/itm/325538949075?hash=item4bcba227d3:g:s3oAAOSwjjxj7w2Y&amdata=enc%3AAQAHAAAAoFZb1f%2Bhh%2B5M5N%2BPe3DYA1I3VxiE7p7TypnF1jia4bsLnkFj0ndiSVQz6QMne5R7064zbuwbtsAH8eiTlY7E6PrnQeVpkNU6kZNKlt8Hx54AYS1wX6lW0TfWU1CgozvEcNS5sZ8Ip20SXUQR6zWHz1I1K1xExktCtXO3vPyDqQw0TkOzfmzMEFscWmuKKpEE98wOjewq6lL1cMBkk8k5kIs%3D%7Ctkp%3ABk9SR-LYkrDNYQ) @ $15

Total expense (already approved) at (15+173+39.99+149.95)=$377.94

# Task Breakdown

The work involved for this would be:

- [x] Purchase UPS hardware
- [ ] Build the UPS to replace the UPS that's there today
- [x] Design/research power system so that 24v power can be powered both off of UPS and off of non backup ports to provide redundancy in case of UPS failure
- [ ] Purchase the new hardware for the PtP being converted to a PtMP radio
- [ ] Preconfigure the RB912 and install it in the stationbox; needs to be able to do OSPF
- [ ] Preconfigure the sector antenna at HIL to be able to do OSPF
- [ ] Install the new radio on CRW and aim the dish to HIL, install UPS
- [ ] Preconfigure LWA client node
- [ ] Collect requirements from OBARC regarding LWA mounting -- what size pipe will it attach to; confirm that equipment on hand for LWA suits this
- [ ] Install LWA client node dish
- [ ] Document that `crw.leb` is abandoned and disable the RF interface

**Comment by turnrye** [comment]
@seichold , if you want to replace the IP cameras at CRW, it'd be good to have that project scheduled for the same site visit.

My availability to work on this at CRW is 2 weeks from now (March 4th or 5th). Beyond that I'm tied up the next few weekends and it would need to wait until April. The power system work could be done separately without any climbing in case it makes sense to split this into two visits.

Purchased the UPS parts just now, total was $414.79 due to sales tax.

**Comment by turnrye** [comment]
My availability to work on this at CRW is 2 weeks from now (March 4th or 5th). Beyond that I'm tied up the next few weekends and it would need to wait until April. The power system work could be done separately without any climbing in case it makes sense to split this into two visits.

**Comment by turnrye** [comment]
Purchased the UPS parts just now, total was $414.79 due to sales tax.

**Comment by turnrye** [comment]
Tracking for the purchases related to the UPS system:
- https://tools.usps.com/go/TrackConfirmAction.action?tLabels=9400108205496495730174
- https://tools.usps.com/go/TrackConfirmAction?qtc_tLabels1=9405508205496493961689
- https://www.fedex.com/fedextrack/?trknbr=394833178028&trkqual=12025~394833178028~FDEG
- https://tools.usps.com/go/TrackConfirmAction?qtc_tLabels1=9449008205496494373213

**Comment by turnrye** [comment]
Just waiting on the rack ears and the battery pack now. There's a risk that the battery pack wont arrive in time -- apparently it got returned to sender and they had to re-ship it yesterday.

**Comment by turnrye** [comment]
Battery pack for UPS is set to arrive today via tracking https://www.fedex.com/fedextrack/?trknbr=395068582443&trkqual=12025~395068582443~FDEG

At this point all of the mail order stuff for the work at Craft Rd and the donation to OBARC has been received. Need to purchase a few incidentals at Home Depot and do the final staging work. A work day has been scheduled for this Saturday for Craft Rd and [announced here](https://discord.gg/wuj4vgXm?event=1079959319493226506).

**Comment by turnrye** [comment]
Modem that we need to convert the dish to a client node shipped this morning, risks not arriving in time for the Saturday work day. Waiting for tracking to become alive to make a decision: 1Z1E2Y740375654283

**Comment by turnrye** [comment]
The battery pack didnt arrive with live batteries, so I had to order more. 1 of the 8 arrived today, and the other 7 are set to arrive tomorrow evening. The work day is being rescheduled now as a result.

**Comment by bobwwj555** [comment]
Ryan what size are the batteries you need is it like in normal APC?

**Comment by turnrye** [comment]
@bobwwj555 it's an RBC43 pack, so the batteries are 12v 5ah sla. Here's an example with the proper physical dimensions: https://www.amazon.com/EXP1250-Terminals-Chamberlain-LiftMaster-Replacement/dp/B0010Z4MDK/ref=sr_1_2?crid=1832H62SO8UYG&dd=tc6mpBsvDbLQ0Ay5aEuoTg%2C%2C&keywords=SP12-5.5HR&qid=1677726276&refinements=p_90%3A8308921011&rnid=8308919011&sprefix=sp12-5.5hr%2Caps%2C177&sr=8-2&qty=6

Even besides the batteries though, the radio wont arrive til tomorrow evening as well. So unfortunately we're going to have to push it back.

**Comment by turnrye** [comment]
UPS config:

address 44.34.129.120/28 (what the PDU used to be on)
gateway 44.34.129.113 (vrrp interface)
